here-naksha-lib-psql/src/main/java/com/here/naksha/lib/psql/PostgresStorage.java

```diff
@@ -450,6 +439,50 @@ synchronized void initStorage(@NotNull PsqlStorage.Params params, @NotNull IoHel
     }
   }
 
+  private void initStoragePlv8(
+      PsqlStorage.@NotNull Params params, @NotNull IoHelp ioHelp, Statement stmt, PsqlConnection conn)
+      throws SQLException {
+    final Plv8Env plv8Env = new Plv8Env();
+    plv8Env.install(conn, latest.toLong());
+    conn.commit();
+  }
+
+  private void initStoragePlpgsql(
+      PsqlStorage.@NotNull Params params, @NotNull IoHelp ioHelp, Statement stmt, PsqlConnection conn)
+      throws SQLException {
+    String SQL;
+    SQL = ioHelp.readResource(ioHelp.findResource("/naksha_plpgsql.sql", PostgresStorage.class));
+    if (logLevel.toLong() >= EPsqlLogLevel.DEBUG.toLong()) {
+      SQL = SQL.replaceAll("--RAISE ", "RAISE ");
+      SQL = SQL.replaceAll("--DEBUG ", " ");
+    }
+    if (logLevel.toLong() >= EPsqlLogLevel.VERBOSE.toLong()) {
+      SQL = SQL.replaceAll("--VERBOSE ", " ");
+    }
+    if (params.pg_hint_plan()) {
+      SQL = SQL.replaceAll("--pg_hint_plan:", " ");
+    }
+    if (params.pg_stat_statements()) {
+      SQL = SQL.replaceAll("--pg_stat_statements:", " ");
+    }
+    SQL = SQL.replaceAll("\n--#", "\n");
+    SQL = SQL.replaceAll("\nCREATE OR REPLACE FUNCTION nk__________.*;\n", "\n");
+    SQL = SQL.replaceAll("\\$\\{schema}", getSchema());
+    SQL = SQL.replaceAll(
+        "\\$\\{version}",
+        logLevel.toLong() > EPsqlLogLevel.OFF.toLong() ? "0" : Long.toString(latest.toLong(), 10));
+    SQL = SQL.replaceAll("\\$\\{storage_id}", storageId);
+    //noinspection SqlSourceToSinkFlow
+    stmt.execute(SQL);
+    conn.commit();
+
+    // Now, we can be sure that the code exists, and we can invoke it.
+    // Note: We do not want to naksha_start_session to be invoked, therefore pass null!
+    initConnection(conn.postgresConnection, null);
+    stmt.execute("SELECT naksha_init();");
```

Comment:
Just a suggestion, Be cautious when executing SQL scripts with stmt.execute. It's generally safer to use stmt.executeBatch or stmt.executeUpdate if the script contains multiple statements, to handle them in a more transactional manner.

Reply:
This code was merely moved around (refactored into a new method), not modified.